circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -138,15 +139,8 @@ trait CirceInstances extends JawnInstances {
       Uri.fromString(str).leftMap(_ => "Uri")
     }
 
-  implicit class JsonDecoderSyntax[F[_]](private val req: Message[F])(implicit F: JsonDecoder[F]) {
-    def asJson: F[Json] = F.asJson(req)
-    def asJsonDecode[A: Decoder]: F[A] = F.asJsonDecode(req)
-  }
-
-  implicit class MessageSyntax[F[_]: Sync](self: Message[F]) {
-    def decodeJson[A](implicit decoder: Decoder[A]): F[A] =
-      self.as(implicitly, jsonOf[F, A])
-  }
+  implicit final def toMessageSynax[F[_]](req: Message[F]): CirceInstances.MessageSyntax[F] =
```

Comment:
Can these functions be called outside the circe package when the syntax class is `private[circe]`?  I thought Scala emitted a warning here about something escaping into a more visible scope.

Reply:
@rossabaker Scala access is somewhat [_"tricky"_](https://github.com/scala/bug/issues/6794).

I believe this should just work, it should be similar to when one uses the **partially applied trick** and the returned class is `private`. I had done that on [neotypes](https://github.com/neotypes/neotypes/blob/master/core/src/main/scala/neotypes/DeferredQuery.scala#L307), and it works without problems.

Also, I just tested this on a console _(inside bloop)_.

```scala
def test[F[_]: cats.effect.Sync](m: org.http4s.Message[F]): F[io.circe.Json] = m.json
// defined function test
```

Correct me if I am wrong, but that was enough proof that the code compiled and AFAIK the Ammonite RELP is outside any package.

-----

However, if you prefer, I could make it public.

Reply:
I'm convinced that it works, even if I am not convinced that it should.